### DISCUSSION TITLE
Netherbot - npcbot tool

### DISCUSSION THREAD

**Discussion by NetherstormX** [trickerer/Trinity-Bots#347] [Show and tell]
Hi,
Just developed a simple addon that helps players play with npcbots. Just want to share it here. Its the first version, will improve and add more functionality to it.
Netherbot is an addon for NPCBots for TrinityCore and AzerothCore 3.3.5, allowing for quick and easy access to various bot-related commands.

![image](https://user-images.githubusercontent.com/94380113/217069606-303cede8-dd48-4c7d-b2bd-4a223747de8f.png)

https://github.com/NetherstormX/NetherBot

@trickerer would be great if you reference it somewhere in the Manual so users know about this and utilize it.

**Comment by trickerer** [comment]
Great job.

I posted a link to your project in WoW repack community discord so people could check it out and give you feedback.

As for me, I immediately noticed 2 things:
1) Bot id to spawn field doesn't get cleared/unfocused after you click `Spawn Bot` button. That would be handy since you're not going to spawn the same bot twice and probably going to move to a different position anyway
2) `Bot-Info` does nothing. Not a bug, I just couldn't find any action assigned to it in the lua.

In general I hope its functionality evolves and includes more advanced stuff in time

**Reply by trickerer** [reply]
One more thing:
There is a general misunderstanding of what bot follow distance is.
For some reason people think it only affects how far bots keep themselves from player which is only its minor function.
Bot follow distance is a distance at which bots acquire/drop their attack target. Actual distance is 72% of the set value for historical reasons.
The point is - default 30 is not **low** value, low value would be 10 to 20 depending on the environment.

I'm saying this because people try to use stand/follow commands to manage engage distance which is way less effective than using follow distance setting.

**Reply by NetherstormX** [reply]
> Great job.
> 
> I posted a link to your project in WoW repack community discord so people could check it out and give you feedback.

Thnx for your support and feedback.

> As for me, I immediately noticed 2 things:
> 
>     1. Bot id to spawn field doesn't get cleared/unfocused after you click `Spawn Bot` button. That would be handy since you're not going to spawn the same bot twice and probably going to move to a different position anyway
Good thinking, now that i tried it it feels annoying. Fixed it immediately.

>     2. `Bot-Info` does nothing. Not a bug, I just couldn't find any action assigned to it in the lua.
 
Thnx for bringing this to my attention. I must have accidentally deleted the function while cleaning up some unused or test code. It has now been resolved. Additionally, clicking the bot will perform the "/bonk" command to retrieve information about the bot. If you select either yourself or another player as the target, it will instead perform the ".npcbot info" command.

**Reply by Conan1996-2** [reply]
I know I'm necroing this thread, but that is a great explanation on the "Follow Distance". I always thought it was how far away from me they would stand and would get upset when they stayed close to me as I looted, when in reality I should be looking at it as an "Attack Distance" by your definition.

**Comment by NetherstormX** [comment]
Just an Update: Raid Frames and Locale Support is added.
![image](https://github.com/trickerer/Trinity-Bots/assets/94380113/9d2943e4-d2a7-4974-8e93-99c734b87c72)

You may update the info that also netherbot can show raid frames.

**Reply by xBarselonx** [reply]
Привет!
Есть аддон который показывает фреймы рейдовой группы, но проблема в том, что он не всегда корректно определяет роль бота при случайном поиске подземелий да при приглашении бота в группу.
Вот ссылка разработчика если интересно - https://github.com/RomanSpector/CompactRaidFrame
